src/devhostagent/PortForward/ServicePortForwardConnector.cs

```diff
@@ -196,25 +197,35 @@ private async Task HookupStreamData(StreamHookData s)
         /// </remarks>
         private byte[] GetModifiedBytesForManagedIdentity(byte[] data)
         {
-            const string headerToInsert = "Metadata: true\n";
-            var stringContent = Encoding.Default.GetString(data.ToList().ToArray());
+            const string headerToInsert = "Metadata: true";
+            var stringContent = Encoding.Default.GetString(data);
+            // This log may not work as expected depending on the character on the string, adding print line inside 
+            // the for lop below to avoid wrong information while debugging
             _log.Verbose($"{loggingPrefix} Original data : {stringContent}");
             stringContent = stringContent.Replace("2017-09-01", "2018-02-01").Replace("clientid", "client_id");
             var lines = stringContent.Split(new char[] { '\n' }).ToList();
             int index = -1;
             for (int i = 0; i < lines.Count; i++)
             {
-                if (lines[i].Contains($"secret: {Common.Constants.ManagedIdentity.SecretValue}"))
+                _log.Verbose($"{loggingPrefix} Printing line : {lines[i]}");
+                if (lines[i].Contains($"secret: {Common.Constants.ManagedIdentity.SecretValue}", StringComparison.OrdinalIgnoreCase))
                 {
                     index = i;
                     break;
                 }
             }
-            lines.Insert(index + 1, headerToInsert);
+            if (index != -1) {
+                lines.Insert(index + 1, headerToInsert);
+            }
+            else {
+                lines.Append(headerToInsert);
+            }
             var modifiedRequestString = string.Join('\n', lines);
 
             _log.Verbose($"{loggingPrefix} Sending data : {modifiedRequestString}");
-            return Encoding.ASCII.GetBytes(modifiedRequestString);
+
+            return Encoding.Default.GetBytes(modifiedRequestString);

```

Comment:
💡☕️ Since I don't know much of background, I will add this thought that - Please note **It is not recommended** to use `Encoding.Default`

* https://learn.microsoft.com/en-us/dotnet/api/system.text.encoding.default?redirectedfrom=MSDN&view=net-7.0#remarks

> Different computers can use different encodings as the default, and the default encoding can even change on a single computer. Therefore, data streamed from one computer to another or even retrieved at different times on the same computer might be translated incorrectly. In addition, the encoding returned by the Default property uses best-fit fallback to map unsupported characters to characters supported by the code page. For these two reasons, using the default encoding is generally not recommended. To ensure that encoded bytes are decoded properly, your application should use a Unicode encoding, such as UTF8Encoding or UnicodeEncoding, with a preamble. Another option is to use a higher-level protocol to ensure that the same format is used for encoding and decoding.

Reply:
this is a good point. Let me look into it. I know ASCII was breaking Go, but let me see if I can UTF8.